### ISSUE TITLE
m3u8 downloader

### ISSUE THREAD

**Issue by nette12** [ytdl-org/youtube-dl#32034]
please help me

for download m3u8 in high speed from RSI ch ?

example: https://rsi-vod-amd.akamaized.net/ch/15587047/6658570695f75a8f330127d146c821f303a63a1af777ce19a14b8612fcb17802/master.m3u8?start=0.0&amp;end=5137.24

**Comment by nette12** [comment]
swiss site RSI

**Comment by dirkf** [comment]
RSI is supported through the extractor/srgssr.py module.
<details><summary>Example: "La storia infinita E01 - Un eterno movimento"</summary>

```console
$ python -m youtube_dl -v -F 'https://www.rsi.ch/play/tv/la-storia-infinita/video/un-eterno-movimento?urn=urn:rsi:video:15733246'
[debug] System config: [u'--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'-F', u'https://www.rsi.ch/play/tv/la-storia-infinita/video/un-eterno-movimento?urn=urn:rsi:video:15733246']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 26035bde4
[debug] Python version 2.7.18 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[SRGSSRPlay] Extracting URL: https://www.rsi.ch/play/tv/la-storia-infinita/video/un-eterno-movimento?urn=urn:rsi:video:15733246
[SRGSSR] Extracting URL: srgssr:rsi:video:15733246
[SRGSSR] 15733246: Downloading JSON metadata
[SRGSSR] 15733246: Downloading HLS-H264-SD token
[SRGSSR] 15733246: Downloading f4m manifest
WARNING: [SRGSSR] Unable to download f4m manifest: HTTP Error 400: Bad Request
[SRGSSR] 15733246: Downloading m3u8 information
WARNING: [SRGSSR] Ignoring subtitle tracks found in the akamai manifest; if any subtitle tracks are missing,; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[SRGSSR] 15733246: Downloading HLS-H264-HD token
[SRGSSR] 15733246: Downloading f4m manifest
WARNING: [SRGSSR] Unable to download f4m manifest: HTTP Error 400: Bad Request
[SRGSSR] 15733246: Downloading m3u8 information
[info] Available formats for 15733246:
format code            extension  resolution note
hls-audio0-Italiano-0  mp4        audio only [it] Italiano 
hls-audio0-Italiano-1  mp4        audio only [it] Italiano 
hls-493-0              mp4        480x272     493k , avc1.640028@ 493k, 24.972fps, video only
hls-493-1              mp4        480x272     493k , avc1.640028@ 493k, 24.972fps, video only
hls-1392-0             mp4        640x360    1392k , avc1.640028@1392k, 24.972fps, video only
hls-1392-1             mp4        640x360    1392k , avc1.640028@1392k, 24.972fps, video only
hls-3592-0             mp4        1280x720   3592k , avc1.640028@3592k, 24.972fps, video only
hls-3592-1             mp4        1280x720   3592k , avc1.640028@3592k, 24.972fps, video only (best)
$ 
```

</details>

In general, downloading an arbitrary M3U8 manifest isn't possible.

Taking the example that you show, when that manifest is fetched by the RSI website (or by yt-dl RSI extractor if it's working correctly), some protocol will have to be followed for the manifest URL to be valid and for that manifest to be accessible. In particular:
* the manifest may only be valid for a limited time
* specific header parameters may need to be sent.

Presumably you know the show that the manifest is supposed to be for. Try the URL for that with yt-dl instead.
